src/bytewords.cpp

```diff
@@ -96,12 +100,25 @@ static inline bool decode_word(const string& word, size_t word_len, uint8_t& out
     return true;
 }
 
+
+#ifdef ESP_PLATFORM
 static inline ByteVector crc32_bytes(const ByteVector &buf) {
     const uint32_t checksum = __builtin_bswap32(esp_crc32_le(0, buf.data(), buf.size()));
     ByteVector result(sizeof(checksum));
     memcpy(result.data(), &checksum, sizeof(checksum));
     return result;
 }
+#else
+static inline ByteVector crc32_bytes(const ByteVector &buf) {
+    uint32_t crc = crc32(0L, Z_NULL, 0);
+    crc = crc32(crc, reinterpret_cast<const Bytef*>(buf.data()), buf.size());
+    const uint32_t checksum = __builtin_bswap32(crc);
+    ByteVector result(sizeof(checksum));
+    memcpy(result.data(), &checksum, sizeof(checksum));
+    return result;
+}
+#endif
+
```

Comment:
Can#t we just have one line conditional, as in the other cases ?  eg:
```suggestion
static inline ByteVector crc32_bytes(const ByteVector &buf) {
#ifdef ESP_PLATFORM
    const uint32_t crc = esp_crc32_le(0, buf.data(), buf.size());
#else
    const uint32_t crc = crc32(crc, reinterpret_cast<const Bytef*>(buf.data()), buf.size());
#endif
    const uint32_t checksum = __builtin_bswap32(crc);
    ByteVector result(sizeof(checksum));
    memcpy(result.data(), &checksum, sizeof(checksum));
    return result;
}
```
or something like that ?